Microsoft.Identity.Web/Resource/AadIssuerValidator.cs

```diff
@@ -71,6 +71,7 @@ public static AadIssuerValidator GetIssuerValidator(string aadAuthority)
                 var aliases = issuerMetadata.Metadata
                     .Where(m => m.Aliases.Any(a => string.Equals(a, authority, StringComparison.OrdinalIgnoreCase)))
                     .SelectMany(m => m.Aliases)
+                    .Append(authority) // For b2c scenarios, the alias will be the authorityHost itself

```

Comment:
**The problem:** Our code is designed to test the issuers against the aliases [login.microsoftonline.com, login.windows.net, etc]. Since AAD authorities always have one of these aliases, we were good for AAD. However B2C have different aliases (even custom ones) and none would be like the aliases above, so this code would reject B2C tokens. So far our reply to this problem was to remove our issuer validator from B2C apps.

**The solution:** I have noticed that we can safely add the current configured authority host as an alias, because a token from yourself should always be valid. So, I added the authority host in the aliases list and with that we opened the possibility to validate b2clogin.com and {myCustomDomain}.com issuers as well.

Reply:
do you have tests for this? we also have sovereign clouds for b2c

---
In reply to: [371542688](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/272#discussion_r371542688) [](ancestors = 371542688)

Reply:
I have added some tests for B2C scenarios on `AadIssuerValidatorTests.cs`. Could you please check if they are ok?